### PR TITLE
clean up and refactor

### PR DESCRIPTION
**Pull request by aniketmaurya**
## What does this PR do?

Few code blocks are rewritten in a more Pythonic way or cleaned

## Before submitting
- [ ] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [ ] Did you write any **new necessary tests**? [not needed for typos/docs]
- [ ] Did you verify **new and existing tests pass** locally with your changes?
- [ ] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/core/data/utils.py [hunk 1]
```diff
@@ -40,7 +40,6 @@
     RunningStage.PREDICTING: "predict",
     RunningStage.SERVING: "serve",
 }
-_STAGES_PREFIX_VALUES = {"train", "test", "val", "predict", "serve"}
```
Comment (aniketmaurya) [inline flash/core/data/utils.py:43]: this is not used anywhere. alternative is `_STAGES_PREFIX_VALUES  = set(_STAGES_PREFIX.values())`

File: flash/core/finetuning.py [hunk 2]
```diff
@@ -170,12 +169,11 @@ def finetune_function(
 
 _FINETUNING_STRATEGIES_REGISTRY = FlashRegistry("finetuning_strategies")
 
-if _PL_AVAILABLE:
-    for strategy in FinetuningStrategies:
-        _FINETUNING_STRATEGIES_REGISTRY(
-            name=strategy.value,
-            fn=partial(FlashBaseFinetuning, strategy_key=strategy),
-        )
+for strategy in FinetuningStrategies:
```
Comment (aniketmaurya) [inline flash/core/finetuning.py:172]: PL is imported directly on the top so this check might be redundant?

File: flash/core/finetuning.py [hunk 3]
```diff
@@ -170,12 +169,11 @@ def finetune_function(
 
 _FINETUNING_STRATEGIES_REGISTRY = FlashRegistry("finetuning_strategies")
 
-if _PL_AVAILABLE:
```
Comment (aniketmaurya) [inline flash/core/finetuning.py:173]: PL is imported directly on the top so this check might be redundant?

File: flash/core/data/io/classification_input.py [hunk 4]
```diff
@@ -65,6 +65,4 @@ def format_target(self, target: Any) -> Any:
         Returns:
             The formatted target.
         """
-        if getattr(self, "target_formatter", None) is not None:
-            return self.target_formatter(target)
-        return target
+        return getattr(self, "target_formatter", lambda x: x)(target)
```
Comment (aniketmaurya) [inline flash/core/data/io/classification_input.py:68]: shall we initialize `target_formatter` with an identity fn?

**Review by ethanwharris — APPROVED** [review]
LGTM 😃

**Pull request merged**